scapy/layers/can.py

```diff
@@ -559,7 +576,11 @@ def read_packet(self, size=CAN_MTU):
         data = data.replace(b' ', b'')
         data = data.strip()
 
-        pkt = CAN(identifier=int(idn, 16), data=binascii.unhexlify(data))
+        if len(data) <= 8:
+            pkt = CAN(identifier=int(idn, 16), data=binascii.unhexlify(data))
```

Comment:
this might be a good time to remove binascii entirely
please use 
``` bytes.fromhex()```
if possible